pages/api/api-mocker/[...slug].ts

```diff
@@ -0,0 +1,199 @@
+import { createClient } from "@supabase/supabase-js";
+import JSON5 from "json5";
+
+const dJSON = require("dirty-json");
+
+import { NextApiRequest, NextApiResponse } from "next";
+import { Database } from "../../../lib/database.types";
+import { RequestMethods } from "../../../lib/models";
+import apiMockPrompt from "../../../lib/prompts/apiMock";
+import { getOpenAIResponse } from "../../../lib/queryOpenAI";
+import { Action } from "../../../lib/types";
+import { exponentialRetryWrapper, splitPath } from "../../../lib/utils";
+
+if (process.env.SERVICE_LEVEL_KEY_SUPABASE === undefined) {
+  throw new Error("SERVICE_LEVEL_KEY_SUPABASE is not defined!");
+}
+
+if (process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY === undefined) {
+  throw new Error("NEXT_PUBLIC_SUPABASE_ANON_KEY is not defined!");
+}
+
+const supabase = createClient<Database>(
+  process.env.NEXT_PUBLIC_SUPABASE_URL ?? "",
+  process.env.SERVICE_LEVEL_KEY_SUPABASE ?? ""
+);
+
+export function slugMatchesPath(path: string, slug: string): boolean {
+  if (splitPath(path).length !== splitPath(slug).length) return false;
+
+  let regexPath =
+    path
+      .replace(/\//g, "\\/")
+      .replace(/{.*?}/g, "([-a-zA-Z0-9@:%_+.~#?&/=]*)") + "?";
+
+  const regex = new RegExp("^" + regexPath + "$");
+
+  const res = regex.test(slug);
+  return res;
+}
+
+export function processMultipleMatches(
+  matches: Action[],
+  slug: string[]
+): Action[] {
+  // slug = 'api/v1/Customers/location'. matches = /api/v1/Customers/location and /api/v1/Customers/{id}.
+  const matchEnd = matches.filter((match) => {
+    const split = match.path?.split("/").filter((arr) => arr != "") ?? [];
+    return split[split.length - 1] === slug[slug.length - 1];
+  });
+  if (matchEnd.length === 1) return matchEnd;
+
+  // slug = 'api/v2/Coordinators/1234'. matches = "/api/v2/Coordinators/{id}". "/api/v2/Coordinators/location"
+  const matchPenultimate = matches.filter((match) => {
+    const split = match.path?.split("/").filter((arr) => arr != "") ?? [];
+    return (
+      split[split.length - 1].includes("}") &&
+      split[split.length - 1].includes("{") &&
+      split[split.length - 2] === slug[slug.length - 2]
+    );
```

Comment:
Does this generalise to all paths?

What if you have `a/path/{like}/{this}`? (Posthog has a few endpoints with multiple path variables)`